README.md

```diff
@@ -16,6 +16,23 @@ This project has adopted the [Microsoft Open Source Code of Conduct](https://ope
 For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or
 contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
 
+## E2E
+This project leverages Terraform and the local user's Azure credentials to run an extensive E2E suite.
+
+### Public Cluster/Public DNS Zone
+The process to run an E2E test for public clusters is as follows: 
+1. Run `make clean-public` or `make clean-all` to clear any preexisting Terraform state for the public cluster dev environment.
+2. Run `make dev-public-cluster` to deploy all Azure resources necessary to run a full suite, including a cluster with the add-on enabled. IMPORTANT: this does not start the add-on. The next step needs to be run for the add-on to be fully deployed with the correct image.
+3. Run `make update-image-on-deployment` to build the add-on image according to the user's local state/branch and push it to the add-on deployment. This step can be re-run when changes to the local add-on are made and the user wishes to manually test those changes on a cluster.
+4. Run `make deploy-e2e` to deploy the e2e tester job, which will run the e2e test suite inside the cluster.
+
+### Public Cluster/Public DNS Zone
+The process to run an E2E test for public clusters is as follows:
```

Comment:
this too seems like it should say private

Reply:
it absolutely should - noticed the typo, but separating this into two PRs, one for updating the public e2e tests/e2e framework, and then a second one for all things private DNS. Will look at the diff once the other is merged in and take care of all changes that need to get done.